sacrifice effect (technique)

JaskRendix opened the pull request:
PR introduces the effect sacrifice.

It was a concept developed by @Sanglorian and it was related to autodestruction types techniques.
These techniques can damage both the user as well as the enemy and they intact the current_hp of the user.
A sacrifice.

M (42/100) - E (69/100).
M uses technique XYZ (with sacrifice effect) against E
`sacrifice 1`
M (0/100) faints - E (27/100).

regarding we can assign the "sacrifice" to assign it to:
- https://wiki.tuxemon.org/Undertaker

we need to change some parameters like type, category, accuracy, potency, power and recharge.

black, isort, tested, no new typehints

Sanglorian commented:
Good idea! It's a useful option to have, and I agree Undertaker's a great fit for it. 

Could the number after sacrifice be a multiplier, so sacrifice 2 would mean "monster takes damage equal to its current HP, and does damage equal to double that amount"?

JaskRendix commented:
> Could the number after sacrifice be a multiplier, so sacrifice 2 would mean "monster takes damage equal to its current HP, and does damage equal to double that amount"?

@Sanglorian done, updated the formula as requested. I added the effect to "undertaker", now it damages and gives the effect prickly; I changed the power (put 1), type (from aether to eart) and range (from special to melee), let me know if it's ok for you.

JaskRendix committed:
sacrifice effect

Sanglorian commented:
> > Could the number after sacrifice be a multiplier, so sacrifice 2 would mean "monster takes damage equal to its current HP, and does damage equal to double that amount"?
> 
> @Sanglorian done, updated the formula as requested. I added the effect to "undertaker", now it damages and gives the effect prickly; I changed the power (put 1), type (from aether to eart) and range (from special to melee), let me know if it's ok for you.

Love it, thanks!
